cmd/proxy/main_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+)
+
+func TestHandleClient_Positive(t *testing.T) {
+	type subtest struct {
+		name    string
+		backend string
+
+		// One stream per query
+		streams []*MockQUICStream
+	}
+
+	subtests := []*subtest{
```

Comment:
Please, define the type inline as an anonymous type. It also doesn't have to be a slice of a pointer type.

```go
tests := []struct{
    name string
    backend string
    streams []*MockQuickStream
}{
    {
        name: "recursive_A",
        // ...
    },
}
```